pkg/reconciler/taskrun/controller.go

```diff
@@ -31,15 +31,36 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 	logger := logging.FromContext(ctx)
 	taskRunInformer := taskruninformer.Get(ctx)
 
+	kubeClient := kubeclient.Get(ctx)
+	pipelineClient := pipelineclient.Get(ctx)
+
+	tsSigner := &chains.TaskRunSigner{
+		Pipelineclientset: pipelineClient,
+		Logger:            logger,
+	}
+
 	c := &Reconciler{
-		TaskRunSigner: &chains.TaskRunSigner{
-			KubeClient:        kubeclient.Get(ctx),
-			Pipelineclientset: pipelineclient.Get(ctx),
-			SecretPath:        SecretPath,
-		},
+		TaskRunSigner: tsSigner,
 	}
+
 	impl := taskrunreconciler.NewImpl(ctx, c, func(impl *controller.Impl) controller.Options {
-		cfgStore := config.NewConfigStore(logger)
+		cfgStore := config.NewConfigStore(logger, func(name string, value interface{}) {
+			// get updated config
+			tsSigner.Config = *value.(*config.Config)
```

Comment:
If the config is invalid, don't continue - log and return (perhaps even shutdown?). We don't want the absence of a config to start causing bad behavior.